shared/src/main/scala/coop/rchain/shared/Log.scala

```diff
@@ -8,12 +8,14 @@ trait LogSource {
 }
 
 object LogSource {
-  def apply(c: Class[_]) = new LogSource {
+  def apply(c: Class[_]): LogSource = new LogSource {
     val clazz: Class[_] = c
   }
 }
 
 trait Log[F[_]] {
+  def isTraceEnabled(implicit ev: LogSource): F[Boolean]
+  def trace(msg: String)(implicit ev: LogSource): F[Unit]
```

Comment:
Ideally, adding those two methods would be a separate commit

Reply:
You are absolutely right. I copied these changes from some other branch. That's why it's all in one commit.